datapage/Datapage.ts

```diff
@@ -75,16 +144,18 @@ export const getDatapageJson = async (
  * see https://github.com/owid/owid-grapher/issues/2121#issue-1676097164
  */
 export const getDatapageGdoc = async (
-    googleDocEditLink: string,
+    googleDocEditLinkOrId: string,
     isPreviewing: boolean,
     publishedExplorersBySlug?: Record<string, ExplorerProgram>
 ): Promise<OwidGdocInterface | null> => {
     // Get the google doc id from the datapage JSON file and return early if
     // none found
-    const googleDocId =
-        getLinkType(googleDocEditLink) === "gdoc"
-            ? getUrlTarget(googleDocEditLink)
-            : null
+    const isPlainGoogleId = gdocIdRegex.exec(googleDocEditLinkOrId)
```

Comment:
Instead of having the full edit urls in the ETL? Yeah I'm pretty sure we want the ids only. But I'm open to passing this explicitly - i.e. have a googleDocId field that can be undefined and a goolgeDocEditLink that can be undefined and then use the former if it is given or the latter if no id is there. What do you think?

Reply:
Okay. 🙂 

In that case I think `googleDocEditLinkOrId` is fine, but it would be nice if we didn't have to write these 6 lines of control flow each time we work with it (assuming we'll have other places we'll ingest it). Maybe a `getGoogleIdFromString: (x: string) => string | undefined` function (which, for adoption's sake, we could also then call in `getUrlTarget` ?)

e.g.

```ts
const googleDocId = getGoogleIdFromString(googleDocEditLinkOrId)
```

and in `getUrlTarget`

```ts
if (url.isGoogleDoc) {
  return getGoogleIdFromString(urlString) as string
}
```

Sorry this is quite hair-splitting stuff! 😬 

I've just spent a decent amount of time trying to be consistent throughout all the Gdocs code so we can call the same couple of utility functions over and over, so I'd like to continue to strive for that. 🙏